internal/graph/dispatch_throttling_check_resolver.go

```diff
@@ -70,30 +76,7 @@ func (r *DispatchThrottlingCheckResolver) GetDelegate() CheckResolver {
 }
 
 func (r *DispatchThrottlingCheckResolver) Close() {
-	r.done <- struct{}{}
-}
-
-func (r *DispatchThrottlingCheckResolver) nonBlockingSend(signalChan chan struct{}) {
-	select {
-	case signalChan <- struct{}{}:
-		// message sent
-	default:
-		// message dropped
-	}
-}
-
-func (r *DispatchThrottlingCheckResolver) runTicker() {
-	for {
-		select {
-		case <-r.done:
-			r.ticker.Stop()
-			close(r.done)
-			close(r.throttlingQueue)
-			return
-		case <-r.ticker.C:
-			r.nonBlockingSend(r.throttlingQueue)
-		}
-	}
+	r.throttler.Close()
```

Comment:
At this point it's clear that all Check resolvers must have the `setDelegate` and `getDelegate` methods. Can you please add them to the interface?

```
type CheckResolver interface {
	ResolveCheck(ctx context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error)
	Close()
	GetDelegate() CheckResolver
	SetDelegate(delegate CheckResolver)
}
```

you should run `make generate-mocks`, too.

Reply:
Agreed, but can I do this as part of the CheckResolver construction PR. It would make more sense to have it there